pkg/controllers/deprovisioning/helpers.go

```diff
@@ -12,25 +12,103 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package consolidation
+package deprovisioning
 
 import (
 	"context"
+	"fmt"
 	"math"
 	"strconv"
 
 	"github.com/samber/lo"
 
 	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
+	nodeutils "github.com/aws/karpenter-core/pkg/utils/node"
 
 	"github.com/aws/karpenter-core/pkg/cloudprovider"
+	"github.com/aws/karpenter-core/pkg/controllers/provisioning"
+	pscheduling "github.com/aws/karpenter-core/pkg/controllers/provisioning/scheduling"
+	"github.com/aws/karpenter-core/pkg/controllers/state"
 	"github.com/aws/karpenter-core/pkg/scheduling"
 
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"knative.dev/pkg/logging"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func simulateScheduling(ctx context.Context, kubeClient client.Client, cluster *state.Cluster, provisioner *provisioning.Provisioner,
```

Comment:
Yeah, I can do that

Reply:
I don't think so, the simulation here uses candidateNodes, which are specific to Deprovisioning, but we would need to reference a provisioner here, so it would be a import cycle.

Reply:
Fair, I guess I'm mainly thinking that this be more bite-sized, if we broke this up a bit so that we have something like

```go
stateNodes := getStateNodes()
pods := getPods()
provisioner.NewScheduler(ctx, pods, stateNodes, pscheduling.SchedulerOptions{
	SimulationMode: true,
})
```